src/modules/CaaSMapper.ts

```diff
@@ -718,10 +793,25 @@ export class CaaSMapper {
     const locale =
       remoteProjectKey && this.api.remotes ? this.api.remotes[remoteProjectKey].locale : this.locale
 
-    const ids = Object.keys(referencedItems)
-    const idChunks = chunk(ids, REFERENCED_ITEMS_CHUNK_SIZE)
-    if (ids?.length > 0) {
-      const response = await Promise.all(
+    const referencedIds = Object.keys(referencedItems)
+
+    const resolvedIds = Object.keys(this.resolvedReferences)
```

Comment:
` const resolvedIds = new Set(Object.keys(this.resolvedReferences))
   const idsToFetchFromCaas = referencedIds.filter(id => !resolvedIds.has(id))
`
Faster and easier to read